src/blob_gc_job.cc

```diff
@@ -353,7 +380,13 @@ bool BlobGCJob::DiscardEntry(const Slice& key, const BlobIndex& blob_index) {
     abort();
   }
 
-  return !(blob_index == other_blob_index);
+  bool relocate = !(blob_index == other_blob_index);
+  if (relocate && metrics) {
+    RecordTick(stats_, BLOB_DB_GC_NUM_KEYS_RELOCATED);
```

Comment:
Even when `relocate == true`, the blob record can be rejected by write callback in `RewriteValidKeyToLSM()`. We should increase `BLOB_DB_GC_NUM_KEYS_RELOCATED` there, only after write success.

Reply:
ditto